Teosinte Expression Models
========================================================

## Set up
```{r}
library("edgeR")
library(ggplot2)
files<-c("265_1A.count","265_2B.count","265_3C.count","265_1A_2.count","265_2B_2.count","265_4A.count","265_1B.count","265_3A.count","265_4B.count","265_2A.count","265_3B.count","265_4C.count","400_1A.count","400_2B.count","400_3C.count","400_1A_2.count","400_2B_2.count","400_4A.count","400_1B.count","400_3A.count","400_4C.count","400_2A.count","400_3B.count")
```

## Load stuff into EdgeR
Counts files are from HTseq analysis of STAR alignment to genome
```{r}
setwd("/Users/jri/projects/dolores/teo_plastic/results/")
counts=readDGE(files)$counts
metadat<-data.frame(files)
metadat$shortname=colnames(counts)
metadat$co2=c(rep(265,12),rep(400,11))
metadat$mom=c("maize","maize","maize","teo","teo","teo","teo","teo","teo","teo","teo","maize","maize","maize","maize","teo","teo","teo","teo","teo","maize","teo","teo")
noint=rownames(counts) %in% c("__no_feature","__ambiguous","__too_low_aQual","__not_aligned","__alignment_not_unique")
cpms<-cpm(counts)
```

Keep only genes with >= 3 hits and calculate normalization factors
```{r}
keep=rowSums(cpms>1) >= 3 & !noint
og.counts<-counts[keep,]
d=DGEList(counts=og.counts,group=metadat$co2)
d2=calcNormFactors(d)
```

## Analysis
Plot MDS of data
```{r fig.width=5, fig.height=5}
plotMDS(d2,pch=16,labels=metadat$shortname,cex=.75,col=c("green","blue")[factor(metadat$co2)])
```

ML analysis of neg. binomial
```{r}
d3=estimateCommonDisp(d2)
d4=estimateTagwiseDisp(d3)
```

Plot mean, variance and CoV
```{r fig.width=5, fig.height=5}
plotMeanVar(d4,show.tagwise.vars=TRUE,NBline=TRUE)
plotBCV(d4)
```

EdgeR analyses of expression diffs.
```{r}
de=exactTest(d4)
tt=topTags(de,n=nrow(d4))   
```

Plot tag mean vs variance
```{r fig.width=5, fig.height=5}
#counts per million of d4
nc=cpm(d4,normalized.lib.sizes=TRUE)
rn=rownames(tt$table)
#highlight stuff with log diff >2 or with FDR < 0.01
deg=rn[tt$table$FDR<0.01]
#deg=rn[tt$table$logFC>2]
plotSmear(d4,de.tags=deg,ylab="log fold counts",main="maize")
abline(h=c(2,-2),col="grey",lwd=2,lty=2)
```

```{r}
write.csv(tt$table,file="toptags_edgeR.csv")
```

Simulation to test differences in means/W stats
```{r}

n=1:23
ustats=as.numeric();
meanstats=as.numeric()
for(i in 1:1000){
  
  test<-c(sample(1:12,4),sample(13:23,4))
  non<-n[!(1:23 %in% test)]

  test<-c(1:3,12:15,21)
  
  mz<-og.counts[,test]
  teo<-og.counts[,non]
  
  d=DGEList(counts=mz,group=metadat$co2[test])
  d2=calcNormFactors(d)
  d3=estimateCommonDisp(d2)
  d4=estimateTagwiseDisp(d3)
  de=exactTest(d4)
  mtt=topTags(de,n=nrow(d4))   
  
  e=DGEList(counts=teo,group=metadat$co2[non])
  e2=calcNormFactors(e)
  e3=estimateCommonDisp(e2)
  e4=estimateTagwiseDisp(e3)
  ee=exactTest(e4)
  ttt=topTags(ee,n=nrow(e4))   
  meanstats[i]=mean(abs(ttt$table$logFC)-abs(mtt$table$logFC))
  ustats[i]=wilcox.test(ttt$table$logFC,mtt$table$logFC)$statistic
  print(i)
}
```

